### PR TITLE
Moved output from Hero to Databox

### PR DESCRIPTION
**Pull request by calebjclark**
+ I renamed @ulixee/testing to @ulixee/hero-testing as I've come to the conclusion that each tool needs its own testing package.

### CODE REVIEW

File: client/lib/Hero.ts [hunk 1]
```diff
@@ -465,8 +452,7 @@ class SessionConnection {
   constructor(private hero: Hero, stateOptions: IStateOptions) {
     const { connectionToCore, ...options } = stateOptions;
 
-    // @ts-ignore
-    const createConnectionToCoreFn: ICreateConnectionToCoreFn = hero.constructor.createConnectionToCore;
+    const createConnectionToCoreFn = (hero.constructor as any).createConnectionToCore;
```
Comment (blakebyrnes) [inline client/lib/Hero.ts:455]: I don't understand this code. Is it just saying `Hero.createConnectionToCore`? You can't change it per instance right?

Comment (calebjclark) [reply client/lib/Hero.ts:455]: Look at fullstack/lib/Hero.ts. It extends the Hero class and sets ConnectionToLocalCore on createConnectionToCore. Therefore, I can't refer to Hero.createConnectionToCore. I must respect the child class.

File: fullstack/index.ts [hunk 2]
```diff
@@ -1,6 +1,5 @@
 import '@ulixee/commons/SourceMapSupport';
 import {
-  Observable,
   ConnectionToRemoteCoreServer,
   ConnectionToCore,
   InteractionCommand,
```
Comment (blakebyrnes) [inline fullstack/index.ts:5]: Lint error a few lines down. On purpose? We should add the eslint hint if so

File: client/lib/Hero.ts (hunk 1 above)
Comment (blakebyrnes) [reply client/lib/Hero.ts:455]: Oh, I see. Should we have a definition on the base class that just doesn't have an implementation? Kind of seems like an abstract variable of sorts.

File: client/lib/Hero.ts [hunk 3]
```diff
@@ -410,6 +409,45 @@ export default class Hero extends AwaitedEventTarget<{ close: void }> implements
...
+    listenerFn: (event: IClassEvents[K]) => any,
+  ): void {
+    return this.removeListener(eventType, listenerFn);
+  }
+
+  public static once<K extends keyof IClassEvents>(
+    eventType: K,
+    listenerFn: (event: IClassEvents[K]) => any,
+  ): void {
+    return this.addListener(eventType, listenerFn);
```
Comment (blakebyrnes) [inline client/lib/Hero.ts:449]: Should this actually implement once?

File: client/lib/Hero.ts (hunk 1 above)
Comment (calebjclark) [reply client/lib/Hero.ts:493]: I added to IHero.

Comment (calebjclark) [reply client/lib/Hero.ts:493]: but yes, I will add.

Comment (calebjclark) [reply client/lib/Hero.ts:493]: Good idea, will add.

File: client/lib/Hero.ts (hunk 3 above)
Comment (calebjclark) [reply client/lib/Hero.ts:450]: Databox won't implement once, but why shouldn't the functionality be there?

File: client/lib/Hero.ts (hunk 1 above)
Comment (calebjclark) [reply client/lib/Hero.ts:494]: Added

File: fullstack/index.ts (hunk 2 above)
Comment (calebjclark) [reply fullstack/index.ts:5]: Fixed, but it's not showing in my Github UI.

File: client/lib/Hero.ts (hunk 3 above)
Comment (blakebyrnes) [reply client/lib/Hero.ts:450]: It just needs to call this.emitter.once... is my point ;)

Comment (calebjclark) [reply client/lib/Hero.ts]: Ah. Good catch. Pushed fix.

**Pull request merged**